### ISSUE TITLE
Code completion on IntelliJ is very slow ~400ms per completion on ARM Macs

### ISSUE THREAD

**Issue by KlausJokisuo** [flutter/flutter-intellij#5825]
## Steps to Reproduce

Generate default Flutter counter project and start typing.

![image](https://user-images.githubusercontent.com/36441457/137601981-8f8665bc-46a7-4cc2-af95-0db97c5c5e1f.png)

## Version info

```
[✓] Flutter (Channel stable, 2.5.2, on macOS 11.6 20G165 darwin-arm, locale en-FI)
    • Flutter version 2.5.2 at /opt/homebrew/Caskroom/flutter/2.5.2/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 3595343e20 (2 weeks ago), 2021-09-30 12:58:18 -0700
    • Engine revision 6ac856380f
    • Dart version 2.14.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/klaus/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 13.0, Build version 13A233
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] IntelliJ IDEA Ultimate Edition (version 2021.2.3)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 61.2.4
    • Dart plugin version 212.5486

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 94.0.4606.81

• No issues found!
```

**Comment by KlausJokisuo** [comment]
@LucasTran380381 Can you show the latency analytics difference between Intel and Native Dark SDK?

![image](https://user-images.githubusercontent.com/36441457/137627004-f5c1b8c4-01c7-420d-9551-81a2591a5b8c.png)

**Comment by LucasTran380381** [comment]
when use native sdk without build in flutter I can not run project. but, I can clearly feel it faster than intel sdk. But I found that if use code dart, just dart file with class not widget, It show completion very fast

**Comment by devoncarew** [comment]
@zanderso @jmagman @christopherfujino - is the Dart SDK built and vended into the Flutter SDK build as x86 or Apple Silicon?

**Comment by stevemessick** [comment]
@devoncarew Is this a dart-lang/sdk issue?

**Comment by devoncarew** [comment]
> @devoncarew Is this a dart-lang/sdk issue?

If the perf issue is that the Dart SDK inside the Flutter SDK isn't built for apple silicon, than the issue should live in the flutter repo. cc @a-siva who has more knowledge of the dart sdk built for apple silicon (and whether the current dart sdk does something like have a fat binary). Steve, there should be an existing issue in the flutter repo for shipping an apple silicon native dart sdk; if you do de-dup this into that issue it would be worth adding info about the impact there (2x completion times).

**Comment by a-siva** [comment]
> If the perf issue is that the Dart SDK inside the Flutter SDK isn't built for apple silicon, than the issue should live in the flutter repo. cc @a-siva who has more knowledge of the dart sdk built for apple silicon (and whether the current dart sdk does something like have a fat binary). Steve, there should be an existing issue in the flutter repo for shipping an apple silicon native dart sdk; if you do de-dup this into that issue it would be worth adding info about the impact there (2x completion times).

No, the current Dart SDK does not build a fat binary, we have one SDK for macOS/X64 and one for macOS/Arm64.

**Comment by KlausJokisuo** [comment]
Any news on this? I've also noticed when the project grows the whole analyzer becomes painfully slow.

**Comment by a-siva** [comment]
> @a-siva I download dart sdk for native arm, then replace folder in flutter folder. When it run
> 
> Launching lib/main.dart on sdk gphone64 arm64 in debug mode... lib/main.dart:1 Snapshot not compatible with the current VM configuration: the snapshot requires 'release no-code_comments no-dwarf_stack_traces_mode lazy_async_stacks lazy_dispatchers use_bare_instructions no-dedup_instructions no-"asserts" "use_field_guards" "use_osr" x64-sysv no-null-safety' but the VM has 'release no-code_comments no- dwarf_stack_traces_mode lazy_async_stacks lazy_dispatchers use_bare_instructions no-dedup_instructions no-"asserts" "use_field_guards" "use_osr" arm64-sysv no-null-safety' 2 the Dart compiler exited unexpectedly.

This error with mismatched snapshot versions is correct. When you manually replace the Dart sdk in the flutter folder you would bring in a different version of the SDK than the one which is used in the flutter engine and hence the version mismatch. Please see https://github.com/flutter/flutter/issues/84453 for the engine specific issue and https://github.com/flutter/flutter/issues/60118 for the overall list of tasks still pending.

**Comment by m-j-g** [comment]
Imagine my disappointment when I setup my new m1 max pro laptop.  Android Studio is painful to use right now.

**Comment by LucasTran380381** [comment]
> Imagine my disappointment when I setup my new m1 max pro laptop. Android Studio is painful to use right now.

Use arm version! I use android studio in my mac mini m1 and it’s insane fast

**Comment by m-j-g** [comment]
> > Imagine my disappointment when I setup my new m1 max pro laptop. Android Studio is painful to use right now.
> 
> Use arm version! I use android studio in my mac mini m1 and it’s insane fast

oh really??  Well, I'm using the Apple Silicon Android Studio install.  Are you talking about replacing the dart folder like discussed earlier?  Sounds like there are issues with that.

**Comment by zanderso** [comment]
https://github.com/flutter/flutter/issues/86926 should help with this.

**Comment by Sunbreak** [comment]
Apple Silicon Dart SDK is available since [flutter-2.11-candidate.7](https://github.com/flutter/flutter/tree/flutter-2.11-candidate.7)

Any update?
